go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -588,7 +589,36 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		}
 	}
 
-	ops, err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bnc.nbClient, nil, ls, lsp)
+	lsps := []*nbdb.LogicalSwitchPort{lsp}
+	if kubevirt.IsPodOwnedByVirtualMachine(pod) && isAllowedForMigration(bnc.IsSecondary(), bnc.IsPrimaryNetwork(), bnc.isLayer2Interconnect()) {
+		kubevirtLiveMigrationStatus, err := kubevirt.DiscoverLiveMigrationStatus(bnc.watchFactory, pod)
+		if err != nil {
+			return nil, nil, nil, false, err
+		}
+		if kubevirtLiveMigrationStatus != nil {
+			if pod.Name == kubevirtLiveMigrationStatus.TargetPod.Name {
+				lsp.Enabled = ptr.To(kubevirtLiveMigrationStatus.State == kubevirt.LiveMigrationTargetDomainReady)
+				sourcePodLspName := bnc.GetLogicalPortName(kubevirtLiveMigrationStatus.SourcePod, nadName)
+				if kubevirtLiveMigrationStatus.State == kubevirt.LiveMigrationTargetDomainReady {
+					// closing the sourcePod lsp to ensure traffic goes to the now ready targetPod.
+					sourcePodLSP := &nbdb.LogicalSwitchPort{
+						Name:    sourcePodLspName,
+						Enabled: ptr.To(false),
+					}
+					lsps = append(lsps, sourcePodLSP)
+				} else if kubevirtLiveMigrationStatus.State == kubevirt.LiveMigrationFailed {
```

Comment:
in failed migration, why do we keep the target LSP around ? Can that be reconciled later on @qinqon ?

Reply:
Did you act on this ?